src/Microsoft.TemplateEngine.Core/Expressions/Cpp/CppStyleEvaluatorDefinition.cs

```diff
@@ -167,30 +172,30 @@ public static bool CppStyleEvaluator(IProcessorState processor, ref int bufferLe
                         {
-                            // this token is a quote - it must match the original quote
-                            if (currentTokenFamily == foundTokenFamily)
-                            {   // we were already in a quoted string, this is the end quote
-                                currentTokenBytes.AddRange(trie.Tokens[token]);  // Add the quote. It's necessary for the evaluator to infer the type - which itself needs work.
-                                string literal = processor.Encoding.GetString(currentTokenBytes.ToArray());
+                            if (inQuoteType == TokenFamily.Whitespace)
+                            {   // starting quote found
+                                currentTokenBytes.AddRange(trie.Tokens[token]);
+                                inQuoteType = foundTokenFamily;
+                            }
+                            else if (foundTokenFamily == inQuoteType)
+                            {   // end quote found
+                                currentTokenBytes.AddRange(trie.Tokens[token]);
                                 tokens.Add(new TokenRef
                                 {
                                     Family = TokenFamily.Literal,
-                                    Literal = literal
+                                    Literal = processor.Encoding.GetString(currentTokenBytes.ToArray())
                                 });
                                 currentTokenBytes.Clear();
-                                // we finished the quoted literal.
-                                // Change this to a value so no processing gets done on an assumption of what this (the previous token) was.
-                                currentTokenFamily = TokenFamily.Whitespace;
+                                inQuoteType = TokenFamily.Whitespace;
                             }
                             else
-                            {
-                                // this is the lead quote
-                                currentTokenFamily = foundTokenFamily;
-                                currentTokenBytes.AddRange(trie.Tokens[token]);  // Add the quote. It's necessary for the evaluator to infer the type - which itself needs work.
+                            {   // this is a different quote type. Treat it like a non-match, just add the token to the currentTokenBytes
+                                currentTokenBytes.AddRange(trie.Tokens[token]);
                             }
                         }
-                        else if (currentTokenFamily == TokenFamily.QuotedLiteral || currentTokenFamily == TokenFamily.SingleQuotedLiteral)
-                        {   // we're inside a quoted literal, the token found by the trie should not be processed, just included with the literal
+                        else if (inQuoteType != TokenFamily.Whitespace)
```

Comment:
Feels like we should have a separate enum for quote types and do the translation back & forth rather than having to use the convention that non-"in quote" situations are whitespace